OSMapsAPI/OpenLayers/map.js

```diff
@@ -42,35 +44,42 @@ function setupLayer() {
                 instructions.classList.remove("hidden");
                 return;
             }
+            // Set correct attribution for the data layer.
             options.attributions = '&copy; <a href="http://www.ordnancesurvey.co.uk/">Ordnance Survey</a>';
 
             var source = new ol.source.WMTS(options);
             var layer = new ol.layer.Tile({ source: source });
-
+            
+            // Error handling should the tiles fail to load. This can be extended to catch specific errors.
             source.on('tileloaderror', function(event) {
                 message.classList.add("warning");
                 message.textContent = 'Could not load a map tile. Ensure you are entering a project API key for a project that contains the OS Maps API';
             });
-
+            
+            // Set up the view options, center of map, zoom level and projection information
             var viewOptions = {
                 projection: options.projection,
                 center: [-121099, 7161610],
                 resolutions: options.tileGrid.getResolutions(),
                 zoom: 10
             }
-
+            
+            // Some additional parameters if we are using a layer in EPSG:3857
             if(options.projection === bng) {
                 var point = new ol.geom.Point(viewOptions.center);
                 point.transform('EPSG:3857', bng);
                 viewOptions.center = point.getCoordinates();
                 viewOptions.zoom = 3;
             }
-
+            
+            // Define the main map element
             map = new ol.Map({
                 target: 'map',
                 layers: [layer],
                 view: new ol.View(viewOptions)
             });
+            
+            // Alter map controls to make sure they don't collapse into each other
```

Comment:
Expand the attribution control, to show the copyright message

Reply:
There's a faceplant smiley missing in the comments box... Of course - we add the copyright message as a collapsed element.